### ISSUE TITLE
Arch Linux: Cannot install/update AUR packages (fs_resolvconf: No such file or directory)

### ISSUE THREAD

**Issue by curiosityseeker** [netblue30/firejail#5723]
### Description

See title of the issue.

### Steps to Reproduce

_Steps to reproduce the behavior_

1. Install or update an AUR package with an AUR helper or via `git clone ....` and `makepkg -s`. 
2. You'll see the following error:
```
==> Validating source files with sha256sums...
    random_package ... Error mount: fs_etc.c:142 fs_resolvconf: No such file or directory
Error: proc XXXXX cannot sync with peer: unexpected EOF
Peer YYYYY unexpectedly exited with status 1
FAILED
==> ERROR: One or more files did not pass the validity check!
```

### Expected behavior

_What you expected to happen_

### Actual behavior

_What actually happened_

### Behavior without a profile

`makepkg` is not sandboxed by default, but `hasher-common.profile` is called.

### Additional context

Commenting `private-etc` in `hasher-common.profile` fixes the issue. I haven't had the time to investigate if a more fine-grained `private-etc` rule can be applied.

### Environment

- Arch Linux
- firejail-git
- fb49f72999a60f7a05575698b6d10e1fa00a39a4

### Checklist

- [x] The issues is caused by firejail. Commenting `private-etc` in `hasher-common.profile` fixes the issue.
- [ ] I can reproduce the issue without custom modifications (e.g. globals.local).
- [ ] The program has a profile. (If not, request one in `https://github.com/netblue30/firejail/issues/1139`)
- [x] The profile (and redirect profile if exists) hasn't already been fixed [upstream](https://github.com/netblue30/firejail/tree/master/etc).
- [ ] I have performed a short search for similar issues (to avoid opening a duplicate).
  - [ ] I'm aware of `browser-allow-drm yes`/`browser-disable-u2f no` in `firejail.config` to allow DRM/U2F in browsers.
- [ ] I used `--profile=PROFILENAME` to set the right profile. (Only relevant for AppImages)

### Log

<details>
<summary>Output of <code>LC_ALL=C firejail /path/to/program</code></summary>
<p>

```
output goes here
```

</p>
</details>

<details>
<summary>Output of <code>LC_ALL=C firejail --debug /path/to/program</code></summary>
<p>

```
output goes here
```

</p>
</details>

**Comment by kmk3** [comment]
On what directory are the AUR repos stored?

On what directory is `makepkg` executed?

**Comment by curiosityseeker** [comment]
That depends. If you [install](https://wiki.archlinux.org/title/Arch_User_Repository#Acquire_build_files) an AUR with `git clone ...` and `makepkg -s` it's really up to you where you want to store it.

If you're using an AUR helper like paru , it is done in `~/.cache/paru`.

**Comment by rusty-snake** [comment]
FTR https://github.com/netblue30/firejail/commit/7176e6324d444b681b822f2a29c15d61c7f92677

They got added to firecfg just recently.

**Comment by curiosityseeker** [comment]
> Yes; I'm asking where they are on _your system_ since that is where the errors are happening. The paths could be currently blocked by an include profile.

Well, as mentioned in the OP, I've tried both AUR helpers (-> ~/.cache) and `git clone` (-> individual paths in ~).

> 
> Also, when adding the following to ~/.config/firejail/sha256sum.local:
> 
> ```
> ignore quiet
> ignore include sha256sum.local
> ignore include globals.local
> ```
> 
> What is the output of `makepkg`?

```
==> Making package: joplin 2.9.17-2 (Thu Mar  9 17:16:17 2023)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> Retrieving sources...
  -> Found joplin.desktop
  -> Found joplin-desktop.sh
  -> Found joplin.sh
  -> Downloading joplin-2.9.17.tar.gz...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 69.3M    0 69.3M    0     0  12.2M      0 --:--:--  0:00:05 --:--:-- 13.1M
==> Validating source files with sha256sums...
    joplin.desktop ... Error mount: fs_etc.c:142 fs_resolvconf: No such file or directory
Error: proc 136337 cannot sync with peer: unexpected EOF
Peer 136338 unexpectedly exited with status 1
FAILED
    joplin-desktop.sh ... Error mount: fs_etc.c:142 fs_resolvconf: No such file or directory
Error: proc 136361 cannot sync with peer: unexpected EOF
Peer 136362 unexpectedly exited with status 1
FAILED
    joplin.sh ... Error mount: fs_etc.c:142 fs_resolvconf: No such file or directory
Error: proc 136386 cannot sync with peer: unexpected EOF
Peer 136387 unexpectedly exited with status 1
FAILED
    joplin-2.9.17.tar.gz ... Error mount: fs_etc.c:142 fs_resolvconf: No such file or directory
Error: proc 136409 cannot sync with peer: unexpected EOF
Peer 136410 unexpectedly exited with status 1
FAILED
==> ERROR: One or more files did not pass the validity check!
```
Again, this does not happen once `private-etc` is commented in `hasher-common.profile`.

**Comment by curiosityseeker** [comment]
Sorry for the noise. The culprit was my `globals.local`. I still don't understand why as it doesn't seem to include any blacklisted paths which look relevant here, though.  But anyway, ignoring it solves the problem.

**Comment by glitsj16** [comment]
> ... Error mount: fs_etc.c:142 fs_resolvconf: No such file or directory

Hmm, longshot, but it _could_ be #5650. Conditions seem to fit the bill, hasher-common.profile does have `net none`. But nobody seems to have been able to reproduce it (yet). I've been using the patch mentioned in #5650 ever since I reported the issue and it WFM on Arch Linux.

**Comment by curiosityseeker** [comment]
Update: I found the culprit. I have 

`dns 127.0.0.1`

in my `globals.local` to make sure that every network request goes through dnscrypt-proxy. Removing that rule fixes the issue although I don't know why. (Note: the blocklists in dnscrypt-proxy didn't block anything related)

**Comment by glitsj16** [comment]
> Update: I found the culprit. I have
dns 127.0.0.1

I'm more and more convinced this is #5650. Can you try the patch from that?

**Comment by curiosityseeker** [comment]
Wow! I applied your patch, removed `ignore dns 127.0.0.1` in hasher-common.local  - and got no errors!

So just to make sure: hasher-common.local now contains 

```
#ignore dns 127.0.0.1
include globals.local

include disable-common.inc
```
 and my `globals.local` contains `dns 127.0.0.1`.

**Comment by kmk3** [comment]
> Duplicate of #5650.

(Closing as a duplicate)
